### ISSUE TITLE
cypher, rule-planner fails matching label

### ISSUE THREAD

**Issue by tbaum** [neo4j/neo4j#4988]
http://console.neo4j.org/r/4cahft

Dataset

```
    CREATE (r:Root),(r)-[:CHILD]->(:ChildType1 { id:"text" }),(r)-[:CHILD]->(:ChildType2 { id:0 })
```

works:

```
    PLANNER COST
    MATCH (:Root)-[:CHILD]->(i:ChildType1)
    WHERE i.id =~ 'te.*'
    RETURN i
```

fails:

```
    PLANNER RULE 
    MATCH (:Root)-[:CHILD]->(i:ChildType1) 
    WHERE i.id =~ 'te.*' 
    RETURN i
```

Error: org.neo4j.cypher.CypherTypeException: Expected 0 to be a java.lang.String, but it was a java.lang.Long

**Comment by thobe** [comment]
Interesting find, it looks like the `RULE` planner schedules the regex filter before the label filter, and fails due to that.

**Comment by Mats-SX** [comment]
Hello @tbaum and thank you for your input. We've addressed this issue and implemented a fix for it, in #5094. Please reopen this issue if the bug is not dealt with by this PR.
